File: Server/src/main/java/pt/ist/ccu2021/queued/Server/dto/StoreDto.java
package pt.ist.ccu2021.queued.Server.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.*;
import pt.ist.ccu2021.queued.Server.domain.Counter;
import pt.ist.ccu2021.queued.Server.domain.Schedule;
import pt.ist.ccu2021.queued.Server.domain.Store;

import java.util.List;
import java.util.stream.Collectors;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Setter
@ToString
@JsonPropertyOrder({"id", "name", "img", "address", "categoryId", "counters", "companyId"})
public class StoreDto {

    @JsonProperty("id")
    private int id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("img")
    private Byte[] img;

    @JsonProperty("address")
    private String address;

    @JsonProperty("categoryId")
    private int categoryId;

    @JsonProperty("counters")
    private List<CounterDto> counters;

    @JsonProperty("schedules")
    private List<ScheduleDto> schedules;

    public StoreDto(Store store, List<CounterDto> counterList, List<Schedule> scheduleList){
        id = store.getId();
        name = store.getName();
        img = store.getImg();
        address = store.getAddress();
        categoryId = store.getCategoryId();
        counters = counterList;
        schedules = scheduleList.stream().map(ScheduleDto::new).collect(Collectors.toList());
    }

    public Store toDomain(int companyId){
        return Store.builder().name(name).img(img).address(address).categoryId(categoryId)
                .companyId(companyId).build();
    }
}

File: Queued/lib/dto/StoreDto.dart
import 'dart:typed_data';
import 'dart:ui';

import 'package:Queued/dto/CounterDto.dart';
import 'package:Queued/dto/ScheduleDto.dart';
import 'package:flutter/cupertino.dart';

class StoreDto {
  int id;

  String name;

  Image img;

  String address;

  int categoryId;

  List<CounterDto> counters;

  List<ScheduleDto> schedules;

  StoreDto.fromJson(Map<String, dynamic> json){
    this.id = json['id'];
    this.name = json['name'];
    this.img = new Image.memory(Uint8List.fromList(json['img'].cast<int>()));
    this.address = json['address'];
    this.categoryId = json['categoryId'];
    List<dynamic> counters = json['counters'];
    this.counters = counters.map((counter) => CounterDto.fromJson(counter)).toList(); //.cast<List<dynamic>>()
    List<dynamic> schedules = json['schedules'];
    this.schedules = schedules.map((schedule) => ScheduleDto.fromJson(schedule)).toList();
  }
}